codice/ddf
# DDF-04426 Expands external host/port properties

#### What does this PR do?
Expands the external host and port properties so that they align with what is displayed in the UI. This makes it so when a user updates the internal host/port in the installer the external ones are not touched.

#### Who is reviewing it? 
@Kjames5269 @jhunzik 

#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@emmberk

#### How should this be tested?
- Run through the installer and try
  - Setting just the internal hostname and ports, verify that custom.system.properties reflects what was set in the UI
  - Setting just the external hostname and ports, verify that custom.system.properties reflects what was set in the UI
  - Setting both external and internal hostname and ports, verify that custom.system.properties reflects what was set in the UI
- Verify ?dev=true still works as expected when setting internal/external ports (functionality should be unchanged)
- Verify [remote headless server instructions](https://github.com/codice/ddf/blob/8e26bed97f7cffa8d3b02b03df63982a6bfab5cf/distribution/docs/src/main/resources/content/_quickstart/quickstart-installing.adoc#quick-install-of-branding-on-a-remote-headless-server) still work (this should not be changed by this change because it is an admin's responsibility to expand/change these properties)
- Verify that updating the hostname and ports in the installer after manually updating custom.system.properties works

#### Any background context you want to provide?
Bug was likely introduced from this commit: https://github.com/codice/ddf/commit/c35076fbd73d3b7abee65dabba721c6004ffabbb

#### What are the relevant tickets?
For GH Issues:
Fixes: #4426 

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
